npm/cmd/start.go

```diff
@@ -212,3 +220,24 @@ func k8sServerVersion(kubeclientset kubernetes.Interface) *k8sversion.Info {
 	}
 	return serverVersion
 }
+
+func labelNode(clientset *kubernetes.Clientset, nodeName, labelValue string) error {
+	msg := fmt.Sprintf("labeling this node %s with %s=%s", nodeName, util.NPMNodeLabelKey, labelValue)
+	metrics.SendLog(util.NpmID, msg, metrics.PrintLog)
+
+	k8sNode, err := clientset.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get k8s node. nodeName: %s. err: %w", nodeName, err)
+	}
+
+	if k8sNode.Labels == nil {
+		k8sNode.Labels = make(map[string]string)
+	}
+	k8sNode.Labels[util.NPMNodeLabelKey] = labelValue
+	_, err = clientset.CoreV1().Nodes().Update(context.TODO(), k8sNode, metav1.UpdateOptions{})
```

Comment:
Could we double-check what happens if some other process updates the node between the `Get` (line 228) and `Update`? I think client-go checks the version and errors here (to avoid overwriting a more recent update), but want to confirm.

Reply:
Looks like `Update()` should have optimistic locking:

https://github.com/kubernetes/client-go/issues/150

https://github.com/kubernetes/client-go/issues/1058#issuecomment-1340234879